packages/hash/api/src/model/ontology/data-type.model.ts

```diff
@@ -59,35 +59,23 @@ export default class {
    *   Depends on the RFC captured by:
    *   https://app.asana.com/0/1200211978612931/1202464168422955/f
    *
-   * @param params.accountId the accountId of the account creating the data type
-   * @param params.schema a `DataType`
+   * @param params.ownedById - the id of the owner of the entity type
+   * @param params.schema - the `DataType`
    */
   static async create(
     graphApi: GraphApi,
     params: {
-      accountId: string;
+      ownedById: string;
       // we have to manually specify this type because of 'intended' limitations of `Omit` with extended Record types:
       //  https://github.com/microsoft/TypeScript/issues/50638
       //  this is needed for as long as DataType extends Record
       schema: Pick<DataType, "kind" | "title" | "description" | "type"> &
         Record<string, any>;
     },
   ): Promise<DataTypeModel> {
-    /** @todo - get rid of this hack for the root account */
-    const namespace =
-      params.accountId === workspaceAccountId
-        ? WORKSPACE_ACCOUNT_SHORTNAME
-        : (
-            await UserModel.getUserByAccountId(graphApi, {
-              accountId: params.accountId,
-            })
-          )?.getShortname();
-
-    if (namespace == null) {
-      throw new Error(
-        `failed to get namespace for account: ${params.accountId}`,
-      );
-    }
+    const namespace = await getNamespaceOfAccountOwner(graphApi, {
+      ownerId: params.ownedById,
+    });
```

Comment:
This code was duplicated across all the ontology model classes, so have encapsulated it in `getNamespaceOfAccountOwner`